src/components/ChannelList/hooks/useChannelListShape.ts

```diff
@@ -353,20 +353,36 @@ export const useChannelListShapeDefaults = <SCG extends ExtendableGenerics>() =>
     async ({
       customHandler,
       event,
+      filters,
       setChannels,
+      sort,
     }: HandleChannelVisibleParameters<SCG>) => {
       if (typeof customHandler === 'function') {
         return customHandler(setChannels, event);
       }
 
-      if (event.type && event.channel_type && event.channel_id) {
-        const channel = await getChannel({
-          client,
-          id: event.channel_id,
-          type: event.channel_type,
-        });
-        setChannels((channels) => uniqBy([channel, ...channels], 'cid'));
+      if (!event.channel) {
+        return;
       }
+
+      const channel = await getChannel({
+        client,
+        id: event.channel.id,
+        type: event.channel.type,
+      });
+
+      const considerArchivedChannels = shouldConsiderArchivedChannels(filters);
```

Comment:
Ah, I think I missed this bit - I could it could happen that a `channel` is archived and hidden so making it visible should respect the archive first, right ?

Reply:
I mean - realistically this should never happen but it's there to respect the filter value in case it ever happens (whatever the actual integration looks like). I just copied and slightly adjusted the `notification.message_new` handler.

Reply:
No no, it's totally on point - I just missed it when doing the fixes 👍